App/components/modals/NotificationModal.tsx

```diff
@@ -0,0 +1,93 @@
+import { useNavigation } from '@react-navigation/core'
+import { StackNavigationProp } from '@react-navigation/stack'
+import React, { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Modal, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
+import Icon from 'react-native-vector-icons/MaterialIcons'
+
+import { Colors } from '../../theme'
+
+import { Button, Title } from 'components'
+import { HomeStackParams } from 'types/navigators'
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: Colors.background,
+  },
+  childContainer: {
+    flexGrow: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  buttonContainer: {
+    justifyContent: 'flex-end',
+    alignItems: 'center',
+  },
+  iconContainer: {
+    width: '100%',
+    flexDirection: 'row',
+    justifyContent: 'flex-end',
+  },
+  iconButton: {
+    padding: 20,
+    paddingVertical: 28,
+  },
+})
+
+interface NotificationModalProps {
+  title: string
+  doneTitle?: string
+  onDone?: () => void
+  onHome?: () => void
+  visible?: boolean
+}
+
+const NotificationModal: React.FC<NotificationModalProps> = ({
+  title,
+  doneTitle,
+  onDone,
+  onHome,
+  visible,
+  children,
+}) => {
+  const { t } = useTranslation()
+  const navigation = useNavigation<StackNavigationProp<HomeStackParams>>()
+  const [modalVisible, setModalVisible] = useState<boolean>(true)
```

Comment:
I wonder if you did `useState<boolean>(visible)` if you would need the `useEffect()` below?

Reply:
Hmm. I didn't think of that but I can give it a try and submit an update